ripc-rxjava1/src/main/java/io/ripc/rx/protocol/tcp/RxConnection.java

```diff
@@ -0,0 +1,64 @@
+package io.ripc.rx.protocol.tcp;
+
+import io.ripc.protocol.tcp.Connection;
+import io.ripc.protocol.tcp.Connection.FlushSelector;
+import rx.Observable;
+import rx.RxReactiveStreams;
+import rx.Subscriber;
+
+import static rx.RxReactiveStreams.*;
+
+/**
+ * An adapter for {@link Connection} representated as an {@link Observable}
+ *
+ * @param <R> The type of objects read from this connection.
+ * @param <W> The type of objects written to this connection.
+ */
+public class RxConnection<R, W> extends Observable<R> {
+
+    private final Connection<R, W> delegate;
+
+    protected RxConnection(final Connection<R, W> delegate) {
+        super(new OnSubscribe<R>() {
+            @Override
+            public void call(Subscriber<? super R> subscriber) {
+                RxReactiveStreams.subscribe(delegate, subscriber);
+            }
+        });
+        this.delegate = delegate;
+    }
+
+    /**
+     * Writes the passed stream of {@code data} and returns the result as an {@link Observable}. All items emitted by
+     * this stream are flushed on completion of the stream.
+     *
+     * @param data Data stream to write.
+     *
+     * @return Result of write.
+     */
+    public Observable<Void> write(Observable<W> data) {
+        return toObservable(delegate.write(toPublisher(data)));
```

Comment:
I think in this case it would actually be more than that since every `Publisher` needs a `Subscriber` and a `Subscription` to be useful. It's likely some objects could be reused but if we assume the worst case (which is probably the most likely) then every write would involve the allocation of at least 5 objects + whatever wrappers, delegates, etc... exist in between (which could mean 3 * numOfTransformations). For Reactor we address this by using a `Stream` to represent input and output (something that's only created once) and we use a `Codec` facility to manage message transformation (something that's also only create once unless not practical).

I was envisioning a "write" facility in this that isn't like `write(byte[])` but like `sink(ByteProducer)`. Namely an operation you could set up once and then pipe data into. You could always use it as a one-off operation but it would be more inefficient in that case.